File: iosApp/iosApp/Presentation/Main/Community/Edit/PostEditUiState.swift
//
//  PostEditUiState.swift
//  ConCafe
//
//  Created by 홍희표 on 4/27/26.
//

import Foundation

struct PostEditUiState {
    var title: String = ""
    var content: String = ""
    var imageUrls: [String] = []
    var imageMaxCount: Int = 5
    var isSubmitting: Bool = false
    var infoMessage: String? = nil
    var isEditMode: Bool = false

    var canSubmit: Bool {
        !title.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty &&
        !content.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
    }
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/community/edit/PostEditUiState.kt
package com.hhp227.concafe.presentation.main.community.edit

data class PostEditUiState(
    val title: String = "",
    val content: String = "",
    val imageUrls: List<String> = emptyList(),
    val imageMaxCount: Int = 5,
    val isEditMode: Boolean = false,
    val isSubmitting: Boolean = false,
    val infoMessage: String? = null
) {
    val canSubmit: Boolean get() = title.isNotBlank() && content.isNotBlank()
}
